Dining/templatetags/dining_tags.py

```diff
@@ -6,21 +6,38 @@
 
 
 @register.filter
-def can_join_slot(slot, user):
+def can_join(dining_list, user):
     # Try creating an entry
-    return slot.can_add_diners(user, check_for_self=True)
+    return dining_list.can_add_diners(user, check_for_self=True)
 
 
 @register.filter
-def is_on_slot(slot, user):
-    return slot.internal_dining_entries().filter(user=user).exists()
+def can_add_others(dining_list, user):
+    """Exhaustive test (for correctness) is not needed since it's only for view usage"""
+    is_adjustable = dining_list.is_adjustable()
+    is_owner = dining_list.is_authorised_user(user)
+    has_room = dining_list.is_open() and dining_list.has_room()
+    limited = dining_list.limit_signups_to_association_only and not user.usermembership_set.filter(
+        association=dining_list.association).exists()
+    return is_adjustable and (is_owner or (has_room and not limited))
```

Comment:
Or just do dining_list.can_add_diners(user) without check for self, should have the same results you try to do here

Reply:
That method is gone

Reply:
To elaborate, there's not a full code duplication with the form clean method and I don't really worry about the minor code duplication because it's for view usage only and therefore has less responsibility of being correct, i.e. doesn't need testcases and also doesn't have to fully conform to the actual business rules.

Have thought about putting the method on the model, but it's likely only used for views and having it on the method means that it does need to be correct, as it may also be used for actual form handling, i.e. it would need test cases and needs to be correct according to the business rules.

Reply:
RIP can_add_diner method. 8/5/19 - 10/5/19

Reply:
Sorry, it was a kind of duplicate of the create new entry form. Have thought of keeping it but then I would've removed the same checks that were in the create form to select the right validation message, to ensure that the business rules are not both in the form and in the model. I've opted for this because the form seemed the best place for business rules

Reply:
I fully support your choice, hence the agree on your earlier post. I just wanted to respect my condolences XD

Reply:
I got it, wanted to bring my condolences as well for killing it. ;p